### ISSUE TITLE
HLLC solver does not work with axisymmetric coords

### ISSUE THREAD

**Issue by zingale** [BoxLib-Codes/Castro#6]
The HLLC solver approximates the flux on the interfaces directly.  For Cartesian coords, the pressure term is part of the momentum flux, so there is no issue.

For axisymmetric coords, this is not the case.  The solution is to either include it in the flux and then have an explicit (non-conservative) geometric source term or ignore it in the flux.  In either case, we still need a valid edge-centered, time-centered pressure.

**Comment by smcouch** [comment]
Have you had a look at what we do in FLASH for HLLC? For the source term case, I'm not sure you need an edge or face centered pressure. I think the zone centered pressure is what you want. And you would want time centered, but you can get away with first order in time for the geometric source terms.....

**Comment by zingale** [comment]
I think that you do want it to be second-order for the update, but I think that can be done with a predictor-corrector.  My main concern though is that HSE will not be preserved in this formulation.  A lesser concern is how to construct the non-conservative terms for the internal energy evolution equation.

**Comment by smcouch** [comment]
Yeah. What we've done in FLASH before is to save the reconstructed and evolved face pressures and average then to get zone and time centered pressures.

**Comment by smcouch** [comment]
I should also say that with Runge-Kutta integrators this is moot because you would just combine the multiple first order updates to get high order.

**Comment by zingale** [comment]
That's a good point, but the timestep restriction is more severe for RK integration, so for the moment we want to avoid that.

**Comment by smcouch** [comment]
I actually think that's a myth, or otherwise only applies to directionally split integrators. I implemented an RK integrator hydro in FLASH and can do 2D Rayleigh Taylor with SSP RK2 at CFL 0.8 and at CFL 1.1 with RK3.
